producer/writer/message_writer.go

```diff
@@ -197,7 +200,7 @@ func (w *messageWriterImpl) Write(rm *producer.RefCountedMessage) {
 	}
 	msg.Set(meta, rm, nowNanos)
 	w.acks.add(meta, msg)
-	w.queue.PushBack(msg)
+	w.queue.PushFront(msg)
```

Comment:
By pushing it to the front, we are able to reuse the scanning logic to just scan the part of new writes and abort, in M3msg’s case there is no guarantee on order because the message could be retried anyway. In real prod use case all the writes within the scan interval will be likely be buffered up and written out in a batch, so eventually they will be written out in approximately the same time

Reply:
It does feel counter-intuitive that we are writing newer messages earlier than older messages, though. Also will this have any impact on which messages will be dropped when the buffer is full?

Reply:
It will not, that is determined by the queue in buffer, which keeps messages in order and drops things in order